oskari-geojson/src/main/java/org/oskari/geojson/GeoJSONReader.java

```diff
@@ -0,0 +1,272 @@
+import org.geotools.feature.DefaultFeatureCollection;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryCollection;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.LinearRing;
+import com.vividsolutions.jts.geom.MultiLineString;
+import com.vividsolutions.jts.geom.MultiPoint;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+
+/**
+ * Convert org.json.JSONObjects (that follow GeoJSON spec) to
+ *  - GeoTools SimpleFeature(Collection)s
+ *  - JTS Geometries
+ */
+public class GeoJSONReader {
+
+    private static final GeometryFactory GF = new GeometryFactory();
+
+    public static SimpleFeatureCollection toFeatureCollection(JSONObject json)
+            throws JSONException {
+        return toFeatureCollection(json, null);
+    }
+
+    public static SimpleFeatureCollection toFeatureCollection(JSONObject json, SimpleFeatureBuilder builder)
+            throws JSONException {
+        String type = json.getString(GeoJSON.TYPE);
+        if (!GeoJSON.FEATURE_COLLECTION.equals(type)) {
+            throw new IllegalArgumentException("type was not " + GeoJSON.FEATURE_COLLECTION);
+        }
+
+        JSONArray features = json.getJSONArray(GeoJSON.FEATURES);
+        final int n = features.length();
+        if (n == 0) {
+            return new DefaultFeatureCollection();
+        }
+
+        if (builder == null) {
+            builder = getBuilder(features.getJSONObject(0));
+        }
+
+        DefaultFeatureCollection fc = new DefaultFeatureCollection();
+        for (int i = 0; i < n; i++) {
+            fc.add(toFeature(features.getJSONObject(i), builder));
+        }
+
+        return fc;
+    }
+
+    public static SimpleFeature toFeature(JSONObject json)
+            throws JSONException {
+        return toFeature(json, null);
+    }
+
+    public static SimpleFeature toFeature(JSONObject json, SimpleFeatureBuilder builder)
+            throws JSONException {
+        String type = json.getString(GeoJSON.TYPE);
+        if (!GeoJSON.FEATURE.equals(type)) {
+            throw new IllegalArgumentException("type was not " + GeoJSON.FEATURE);
+        }
+
+        if (builder == null) {
+            builder = getBuilder(json);
+        }
+
+        builder.reset();
+
+        JSONObject geometry = json.optJSONObject(GeoJSON.GEOMETRY);
+        if (geometry != null) {
+            builder.set("geom", toGeometry(geometry));
```

Comment:
Add "geom" as a constant as it is repeated multiple times on the code. Maybe even use builder.getDefaultGeometry() instead of "geom" here?

Reply:
Changed it to a public static final field. Also changed the value from `"geom"` to `"geometry"` since that's what the GeoJSON plugin in GeoTools calls it.